services/etcd.go

```diff
@@ -318,6 +324,53 @@ func RunEtcdSnapshotSave(ctx context.Context, etcdHost *hosts.Host, prsMap map[s
 	return nil
 }
 
+func DownloadEtcdSnapshot(ctx context.Context, etcdHost *hosts.Host, prsMap map[string]v3.PrivateRegistry, etcdSnapshotImage string, name string, once bool, es v3.ETCDService) error {
+	log.Infof(ctx, "[etcd] Get snapshot [%s] on host [%s]", name, etcdHost.Address)
+	s3Backend := es.BackupBackend.S3BackupBackend
+	if len(s3Backend.Endpoint) == 0 || len(s3Backend.AccessKeyID) == 0 || len(s3Backend.SecretAccesssKey) == 0 || len(s3Backend.BucketName) == 0 {
+		log.Warnf(ctx, "[etcd] Failed to get snapshot [%s] from s3 on host [%s], invalid s3 configurations", name, etcdHost.Address)
+		return nil
+	}
+	imageCfg := &container.Config{
+		Cmd: []string{
+			"/opt/rke-tools/rke-etcd-backup",
+			"etcd-backup",
+			"download",
+			"--name", name,
+			"--endpoints=" + etcdHost.InternalAddress + ":2379",
+		},
+		Image: etcdSnapshotImage,
+	}
+
+	log.Infof(ctx, "invoking restore s3, bucketName=%s, endpoint:%s", s3Backend.BucketName, s3Backend.Endpoint)
```

Comment:
this code looks similar to the code block on lines 280-286, any chance we can extract it and re-use?